addons/markers/functions/fnc_onButtonClickConfirm.sqf

```diff
@@ -0,0 +1,42 @@
+#include "script_component.hpp"
+#include "\a3\ui_f\hpp\defineResincl.inc"
+/*
+ * Author: Freddo
+ * When the confirm button is pressed.
+ *
+ * Arguments:
+ * 0: Confirm button <CONTROL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_buttonOk] call ACE_markers_fnc_onButtonClickConfirm
+ *
+ * Public: No
+ */
+params ["_buttonOk"];
+
+private _display = ctrlParent _buttonOk;
+private _description = _display displayctrl IDC_INSERT_MARKER;
+private _aceTimestamp = _display displayCtrl IDC_ACE_INSERT_MARKER_TIMESTAMP;
+
+// Handle timestamp
+if (cbChecked _aceTimestamp && {[ACE_player] call FUNC(canTimestamp)}) then {
+    private _time = daytime;
+    private _ampm = switch (true) do {
+        case (GVAR(timestampHourFormat) == 24): {""};
+        case (_time < 12): {" am"};
+        case (_time > 12): {SUB(_time,12); " pm"};
+    };
+
+    _description ctrlSetText format [ // Add timestamp suffix
+        "%1%2[%2%3]",
```

Comment:
This doesn't seem right?
description, space, space, time

you give 4 parameter but only use 3, and the space seems wrong

Reply:
Yeah looks like I made a typo, should probably be `"%1%2[%3%4]"`

Reply:
Yes, am-pm is completely broken in this pull request. Even if this part was repaired, am-pm will not be shown for 12 and 24-hours, and even if that part is repaired, it will show the wrong one.

Should all be fixed here: https://github.com/acemod/ACE3/pull/8034/files#diff-4b395ce5b5e0cc52ea30d46cdc2cb9d5fea1eb22d6ca4cfe885b60ef68b1e45eR48